### ISSUE TITLE
REST: Updating a customer without store_id sets the store_id to default 

### ISSUE THREAD

**Issue by 808brinks** [magento/magento2#22869]
### Preconditions (*)
Magento 2.2.7 and Magento 2.3-develop provided by the bot.

### Steps to reproduce (*)
1. Create multiple store views
2. Create a customer and set his store view to one that is not the default
3. Use the rest api to change the first name of the customer (put to PUT rest/V1/customers/8)

I used the following body. (note the missing store_id)

```
{ "customer":         {
            "id": 1,
            "group_id": 1,
            "default_billing": "0",
            "default_shipping": "0",
            "created_at": "2019-05-13 14:33:11",
            "updated_at": "2019-05-13 14:33:11",
            "created_in": "alternate store view",
            "email": "xxx@redacted.com",
            "firstname": "firstname alternate",
            "lastname": "lastname",
            "gender": 0,
            "website_id": 1,
            "addresses": [],
            "disable_auto_group_change": 0
        }
}
```

### Expected result (*)
1. The first name is changed.
2. All fields not passed in the PUT call are not changed. (Including store_id).

```
{
    "items": [
        {
            "id": 1,
            "group_id": 1,
            "default_billing": "0",
            "default_shipping": "0",
            "created_at": "2019-05-13 14:33:11",
            "updated_at": "2019-05-13 14:40:56",
            "created_in": "alternate store view",
            "email": "bob.brinks@nl.spotler.com",
            "firstname": "firstname alternate",
            "lastname": "lastname",
            "gender": 0,
            "store_id": 2,
            "website_id": 1,
            "addresses": [],
            "disable_auto_group_change": 0
        }
    ],
    "search_criteria": {
        "filter_groups": [],
        "page_size": 10
    },
    "total_count": 1
}
```

### Actual result (*)
1. The store_id is changed to the default store_id

```
{
    "items": [
        {
            "id": 1,
            "group_id": 1,
            "default_billing": "0",
            "default_shipping": "0",
            "created_at": "2019-05-13 14:33:11",
            "updated_at": "2019-05-13 14:40:56",
            "created_in": "alternate store view",
            "email": "bob.brinks@nl.spotler.com",
            "firstname": "firstname alternate",
            "lastname": "lastname",
            "gender": 0,
            "store_id": 1,
            "website_id": 1,
            "addresses": [],
            "disable_auto_group_change": 0
        }
    ],
    "search_criteria": {
        "filter_groups": [],
        "page_size": 10
    },
    "total_count": 1
}
```

**Comment by 808brinks** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @808brinks. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @808brinks, here is your [Magento instance](https://i-22869-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-22869-2-3-develop.instances.magento-community.engineering/admin](https://i-22869-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by 808brinks** [comment]
@m2-assistant I was able to reproduce the issue on the provided magento 2 instance.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @AlexWorking 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99688`, `MAGETWO-99689` were created

**Issue Available**: @AlexWorking, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @808brinks. Thank you for your report. 
The issue has been fixed in magento/magento2#22895 by @Wirson in 2.2-develop branch
Related commit(s):
 - [9fd7722f66c5024e98a5d3ade1245f253d890369](https://github.com/magento/magento2/commit/9fd7722f66c5024e98a5d3ade1245f253d890369)
 - [f84f0dba7192a5562f26a85c4896cf2de4cf8be3](https://github.com/magento/magento2/commit/f84f0dba7192a5562f26a85c4896cf2de4cf8be3)

The fix will be available with the upcoming 2.2.10 release.
